src/core/Akka.Persistence/AtLeastOnceDeliverySemantic.cs

```diff
@@ -346,31 +367,62 @@ public AtLeastOnceDeliverySemantic(IActorContext context, PersistenceSettings.At
         /// this helps prevent an overwhelming amount of messages to be sent at once.
         /// 
         /// The default value can be configure with the 'akka.persistence.at-least-once-delivery.redelivery-burst-limit'
-        /// configuration key. This method can be overridden by implementation classes to return
-        /// non-default values.
+        /// configuration key.
         /// </summary>
-        public virtual int RedeliveryBurstLimit => _settings.RedeliveryBurstLimit;
+        public int RedeliveryBurstLimit { get; set; }
 
         /// <summary>
         /// After this number of delivery attempts a <see cref="UnconfirmedWarning" /> message will be sent to
         /// <see cref="ActorBase.Self" />. The count is reset after restart.
         /// 
         /// The default value can be configure with the 'akka.persistence.at-least-once-delivery.warn-after-number-of-unconfirmed-attempts'
-        /// configuration key. This method can be overridden by implementation classes to return
-        /// non-default values.
+        /// configuration key.
         /// </summary>
-        public virtual int WarnAfterNumberOfUnconfirmedAttempts => _settings.WarnAfterNumberOfUnconfirmedAttempts;
+        public int WarnAfterNumberOfUnconfirmedAttempts
+        {
+            get => _warnAfterNumberOfUnconfirmedAttempts;
+            set
+            {
+                if (_warnAfterNumberOfUnconfirmedAttempts != value)
+                {
+                    _warnAfterNumberOfUnconfirmedAttempts = value;
+
+                    var deliveriesAboveThreshold = _unconfirmed.Where(u => u.Value.Attempt >= value)
+                        .Select(u => new UnconfirmedDelivery(u.Key, u.Value.Destination, u.Value.Message))
+                        .ToArray();
+
+                    if (deliveriesAboveThreshold.Length > 0)
+                    {
+                        _context.Self.Tell(new UnconfirmedWarning(deliveriesAboveThreshold));
```

Comment:
Same here, the last thing I would expect from using (`this.WarnAfterNumberOfUnconfirmedAttempts = 5`) is that it may send messages to actors and allocate extra arrays.

Reply:
Again - this is to make the logic correct - if you start with 10, have 6 unconfirmed, then switch to 5 then you should receive the warning.

What do you think is best? Have this read only (and accept the value only via `ctor`?) or not trigger immediately but wait for next unconfirmed message?

Reply:
Agree with @Horusiath - this should be immutable I think. Pick your strategy and stick with it at the time the actor starts.